index.html

```diff
@@ -35,58 +41,150 @@
+              optional TCPSocketOptions options = {});
+
+          readonly attribute Promise&lt;TCPSocketConnection> connection;
+          readonly attribute Promise&lt;undefined> closed;
+
+          Promise&lt;undefined> close();
+        };
+      </pre>
+
+      <p>
+      Methods on this interface typically complete asynchronously, queuing work on the <dfn>TCP socket task source</dfn>.
+
+      <p>
+      Instances of {{TCPSocket}} are created with the internal slots described in the following table:
+
+      <table class="simple" data-dfn-for="TCPSocket">
+        <tr>
+          <th>Internal slot
+          <th>Initial value
+          <th>Description (non-normative)
+        <tr>
+          <td><dfn>[[\readable]]</dfn>
+          <td>`null`
+          <td>A {{ReadableStream}} that receives data from the socket 
+        <tr>
+          <td><dfn>[[\writable]]</dfn>
+          <td>`null`
+          <td>A {{WritableStream}} that transmits data to the socket
+        <tr>
+          <td><dfn>[[\connectionPromise]]</dfn>
+          <td>`null`
+          <td>A {{Promise}} used to wait for the connection to be established. Corresponds to the {{TCPSocket/connection}} member.
+        <tr>
+          <td><dfn>[[\closedPromise]]
+          <td>`null`
+          <td>A {{Promise}} used to wait for the socket to close or error. Corresponds to the {{TCPSocket/closed}} member.
+      </table>
+
+      <section>
+        <h3><dfn>constructor()</dfn> method</h3>
+
+        <aside class="example">
+          In order to communicate via TCP a socket connection must be requested first. The socket object constructor
+          allowes the site to specify the necessary parameters which control how data is transmitted and received.
+
+          <pre class="js">
+            const socket = new TCPSocket(/*remoteAddress=*/, /*remotePort=*/, /*options=*/);
+          </pre>
+
+          The developer should wait for the connection to be established before accessing the {{TCPSocket/connection}} attribute. 
+
+          <pre class="js">
+            const { readable, writable } = await socket.connection;
+          </pre>
+
+          Once {{TCPSocket/connection}} has resolved, the {{TCPSocketConnection/readable}} and {{TCPSocketConnection/writable}}
+          can be accessed to get the {{ReadableStream}} and {{WritableStream}} instances for receiving data from and sending data to the connected socket.
+        </aside>
+
+        The {{TCPSocket/constructor()}} steps are:
+
+        <ol>
+          <li>If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+        not [=allowed to use=] the [=policy-controlled feature=] named "[=policy-controlled feature/direct-sockets=]", throw a
+        "{{NotAllowedError}}" {{DOMException}} and return `null`.
+          <li>If only one of |options|["{{TCPSocketOptions/localAddress}}"] or |options|["{{TCPSocketOptions/localPort}}"] is specified, throw a {{TypeError}}  and return `null`.
```

Comment:
You don't need to return when throwing an exception. If anything, you can just say "and abort these steps" but I don't think even that is necessary as the control flow is clear from context.

Reply:
Fixed!